src/rvsdg/optimize_direct_jumps.rs

```diff
@@ -88,6 +90,24 @@ impl SimpleCfgFunction {
             resulting_graph.add_edge(source, target, edge.weight().clone());
         }
 
+        let mut label_mapping = HashMap::<String, String>::new();
+        for (old, new) in node_mapping.iter() {
+            assert!(
+                label_mapping
+                    .insert(
+                        self.graph[*old].name.to_string(),
+                        resulting_graph[*new].name.to_string(),
+                    )
+                    .is_none(),
+                "Duplicate labels in graph"
+            );
```

Comment:
nit: I know it's technically correct but stateful expressions in asserts is sus.

Reply:
fixed